File: projects/hanoi_learn_python/src/hanoi_learn_python/tour/control_flow.py
#!/usr/bin/env python3



def control_flow():

    # condition
    x = 10
    if x > 5:
        print("x > 5")
    elif x == 5:
        print("x = 5")
    else:
        print("x < 5")

    # fmt: off
    s1 = 4
    g1 =     "A" if s1 > 9 \
        else "B" if s1 > 8 \
        else "C" if s1 > 7 \
        else "?"
    # fmt: on
    print(g1)

    # while loop
    count = 0
    while True:
        count += 1
        if count == 5:
            break
    print(count)

    # type casting
    tc_1_1 = int(10.5)
    tc_1_2 = float(10)

    try:
        tc_2 = int("10")
    except ValueError:
        tc_2 = None

    print(tc_2)  # None

File: projects/hanoi_learn_rust/src/hanoi_learn_rust/tour/control_flow.rs
use std::collections::HashMap;
use crate::hanoi_utils_rust::dbgq;

pub fn control_flow() {
    // for loop
    let num_dict: HashMap<&str, Vec<i32>> = [
        ("prime", vec![2, 3, 5, 7, 11]),
        ("fibon", vec![1, 1, 2, 3, 5]),
        ("squar", vec![1, 4, 9, 16]),
    ]
        .iter()
        .cloned()
        .collect();

    for (ntype, nums) in &num_dict {
        for num in nums {
            dbgq!(ntype, num);
        }
    }

    for i in 0..5 {
        println!("{}", i);
    }

    for i in (7..=10).rev().step_by(2) {
        println!("{}", i);
    }

    // while
    let mut count = 0;
    loop {
        count += 1;
        if count == 5 {
            break;
        }
    }
    println!("count: {}", count);

    // type casting
    let tc_1_1 = 10.5 as i32;
    let tc_1_2 = 10 as f32;

    let tc_2 = match "hello".parse::<i32>() {
        Ok(num) => Some(num),
        Err(_) => None,
    };

    println!("{:?}", tc_2);
}
